rpmio/rpmlog.c

```diff
@@ -237,6 +237,89 @@ const char * rpmlogLevelPrefix(rpmlogLvl pri)
     return prefix;
 }
 
+static const char * rpmlogLevelColor(rpmlogLvl pri)
+{
+    return rpmlogMsgPrefixColor[pri&0x7];
+}
+
+static int rpmlogDefault(FILE *stdlog, rpmlogRec rec)
+{
+    static const char fubar[] =
+	"Error occurred during writing of a log message";
+    FILE *msgout = (stdlog ? stdlog : stderr);
+    const char * colorOn = isatty(fileno(msgout))
```

Comment:
RPM output is often useless in a pager because of inconsistent usage between stdout/stderr while logging. Complicated operations (-Uvh, -ba) that run many scripts cannot colorize without applying patterns using sed.

So adding a --color=WHEN option (instead of a simple colorization of the the log prefix) is likely not worth the effort.

BTW, adding comments after C statements, not in the midst of a statement, provides better context. The snippet above appears to be coercing an integer return into a const char * pointer.